usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -3083,8 +3096,14 @@ arc_evict_state(arc_state_t *state, uint64_t spa, int64_t bytes,
 		 * A b_spa of 0 is used to indicate that this header is
 		 * a marker. This fact is used in arc_adjust_type() and
 		 * arc_evict_state_impl().
+		 * A b_lsize of 0 is used to indicate that we first only
+		 * evict headers which have L2 headers.
+		 * A b_lsize of 1 does the same but tries to find more
+		 * such headers ignoring zfs_arc_evict_batch_limit.
+		 * A b_lsize of 2 catches both L1 & L2 headers (default).
 		 */
 		markers[i]->b_spa = 0;
+		markers[i]->b_lsize = l;
```

Comment:
Yes you're right, I initially did not want to change the function signature to be as less intrusive as possible, but of course it's much nicer. Done !